cms/src/test/java/uk/nhs/digital/apispecs/IsAnyTrueHelperTest.java

```diff
@@ -107,21 +109,39 @@ public void throwExceptionWhenCollectionIsNull() throws IOException {
     }
 
     @Test
-    public void throwExceptionWhenCollectionIsNotBoolean() throws IOException {
+    public void throwExceptionWhenAtLeastOneIsNull() throws IOException {
```

Comment:
As mentioned in earlier comments and discussed on Slack, this test should fail when the method under test (`isAnyTrueHelper.apply`) fails to throw any exceptions (and the one we're asserting against in particular).

Reply:
They are failing. They don't need to be failed explicitly.

Reply:
They are passing if the code of the method under test is modified so that it doesn't throw _any_ exceptions (not just the validations ones).

If you modify the method under test by embracing its _entire_ logic (including the validations) within this try-catch block, you'll see that this test and the `throwExceptionWhenAtLeastOneIsNonBoolean` one will pass.
```
try {
    ...all logic that could throw an exception is placed here...
} catch (Exception e) {
    e.printStackTrace();
}
```

Unlikely as it seems, this example is just one way to make the method not throw _any_ exceptions, but it's a common mistake made by developers who don't have an understanding of how exception handling should work.

The simple call to `fail()` method (like the one in the `throwExceptionWhenCollectionIsEmpty`) protects the code from this kinds of mistakes from occurring.

Reply:
Fixed